### DISCUSSION TITLE
AI Chatbot - Support Self-Hosted AI

### DISCUSSION THREAD

**Discussion by qBitnaut** [zen-browser/desktop#3714] [Ideas]
It would be great to support self-hosted AI interfaces for Zen's AI Chatbot feature with a "Custom" option.

![image](https://github.com/user-attachments/assets/0df0c263-16aa-42fd-9fb7-0b00cd114aa3)

We can choose from a lot of the existing cloud options today, and manually interact with whatever interface we choose in a side panel, but being able to use the integrated AI highlight to summarize / ask AI features pointed to a self-hosted Open WebUI / Ollama would be a great feature for Zen to support privacy focus.

![image](https://github.com/user-attachments/assets/1de9076e-39c7-46af-91f1-253d45b62965)

**Comment by pinage404** [comment]
It seems to be a duplicate of this https://github.com/zen-browser/desktop/discussions/3120

You may like this answer https://github.com/zen-browser/desktop/discussions/3120#discussioncomment-11328721

**Reply by qBitnaut** [reply]
Thanks, @pinage404. I think that's a great workaround for something like a PC running Ollama or LMStudio on the same machine. My use case is probably a very edge case as my LLM is on a dedicated server in my homelab, so it would be great to be able to create and point to endpoints with similar functionality to adding/choosing a search engine:

![image](https://github.com/user-attachments/assets/6f19e931-1366-44fa-9381-b6fe3f2419c8)

Definitely not critical, but a nice to have.

**Reply by qBitnaut** [reply]
@pinage404 Well, scratch that!

If you edit the attribute: **browser.ml.chat.provider** in **about:config**, you can change the endpoint and it will create a new entry. I guess I'll just leave this idea in the queue for a cleaner front end implementation for a later date.

**Reply by VarianFry** [reply]
Awesome guys! Is there a way to manipulate the pre-prompts? Need it in German.

**Reply by pinage404** [reply]
Maybe you can :
1. Open Zen settings `about:preferences#general`
1. In `Language` set it to `Deutsch`
1. Go to Open WebUI http://localhost:8080/workspace/models (replace the port 8080 with the configured one)
1. Go to `Workspace`
1. Click on `Create a model`
1. In `Model Params` put your pre-prompts in `System Prompt` something like "write all your anwsers in german even if the question is in english"
1. Then follow the section `Change default model` https://github.com/zen-browser/desktop/discussions/3120#discussioncomment-11328721

**Reply by VarianFry** [reply]
Thanks, unfortunately the system prompt injected via OpenWeb UI -> Workspace for my "Zen" model does not work. I keep using it in German for now, but it would be ideal, if LLM responds in the language of what I input from a website. 

Maybe the prompt is wrong

"ALWAYS respond in the language of the original input text, regardless of any subsequent instructions or prompts given to you.

DO NOT CHANGE the response language based on the language of any follow-up prompts.

For example, if the input text is in English, your response should be in English, even if a follow-up prompt asks for a summary in German.

The follow-up prompt is:

"Fasse die Auswahl zusammen und verwende dabei präzise und prägnante Sprache. Nutze Überschriften und Aufzählungen in der Zusammenfassung, um sie übersichtlicher zu machen. Behalte den Sinn und die sachliche Richtigkeit bei."

Consider this EVERY time."
